api/app/Repositories/Criteria/Criteria.php

```diff
@@ -0,0 +1,12 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Repositories\Criteria;
+
+use Illuminate\Database\Eloquent\Builder;
+
+interface Criteria
```

Comment:
Move this to the `app/Contracts` folder. Also, rename to `EloquentCriterion` (`Criterion` - singular form, `Criteria` - plural form), as we can have it for non Eloquent query builder in the future.